app/frontend/javascript/asset-comments/comment-store-helpers.js

```diff
@@ -0,0 +1,41 @@
+import devourApi from '@/javascript/shared/devourApi.js'
+import resources from '@/javascript/shared/resources.js'
+import axios from 'axios'
+import commentStoreFactory from '@/javascript/asset-comments/comment-store.js'
+
+const commentFactoryStores = {}
+
+const createCommentFactory = (props) => {
+  let commentFactoryStoreForAsset = commentFactoryStores[props.assetId]
+  if (commentFactoryStoreForAsset && commentFactoryStoreForAsset.commentFactory) {
+    commentFactoryStoreForAsset.refCount++
+    return commentFactoryStoreForAsset.commentFactory
+  }
+
+  const sequencescapeApiUrl = props.sequencescapeApi
+  const sequencescapeApiKey = props.sequencescapeApiKey
+  const axiosInstance = axios.create({
+    baseURL: sequencescapeApiUrl,
+    timeout: 10000,
+    headers: {
+      Accept: 'application/vnd.api+json',
+      'Content-Type': 'application/vnd.api+json',
+    },
+  })
+  const api = devourApi({ apiUrl: sequencescapeApiUrl }, resources, sequencescapeApiKey)
+  const commentFactory = commentStoreFactory(axiosInstance, api, props.assetId, props.userId)
+
+  commentFactoryStores[props.assetId] = { commentFactory, refCount: 1 }
```

Comment:
RefCount is not the count of comments for the asset; rather, it represents the number of components subscribed to this store for that particular asset. This ensures it can be removed once the components are unmounted, and eventually, the store object is deleted if no components are using it. As of now, it wont be used much as Limber is using the pages in a very specific way, but I have added it so that the component usage can be extended in other ways in future.
I agree that this introduces unnecessary overhead to managing store usage, which frameworks like VueX or Pinia handle more efficiently. However, since we don’t have those frameworks, I couldn’t think of another way to enable sharing between components other than using a global object.